test/src/unit-backwards_compat.cc

```diff
@@ -1363,4 +1357,57 @@ TEST_CASE(
   vfs.remove_dir(get_fragment_dir(array_read2.uri()));
   vfs.remove_dir(get_commit_dir(array_read2.uri()));
   vfs.remove_dir(schema_folder);
-}
\ No newline at end of file
+}
+
+template <class T>
+void assert_group_metadata(
+    Group& g,
+    const std::string& key,
+    tiledb_datatype_t expected_datatype,
+    T expected_value) {
+  tiledb_datatype_t datatype;
+  uint32_t value_num;
+  const void* value;
+
+  g.get_metadata(key, &datatype, &value_num, &value);
+  REQUIRE(datatype == expected_datatype);
+  REQUIRE(value_num == 1);
+  REQUIRE(value != nullptr);
+  REQUIRE(*static_cast<const T*>(value) == expected_value);
+
+  std::vector<T> expected_values(7);
+  std::fill(expected_values.begin(), expected_values.end(), expected_value);
+  g.get_metadata(key + "_multi", &datatype, &value_num, &value);
+  REQUIRE(datatype == expected_datatype);
+  REQUIRE(value != nullptr);
+  const T* valueT = static_cast<const T*>(value);
+  REQUIRE(std::vector<T>(valueT, valueT + value_num) == expected_values);
+}
+
+TEST_CASE(
+    "Backwards compatibility: Test reading group metadata written with "
+    "previous version of tiledb",
+    "[backwards-compat][group][metadata]") {
+  Context ctx;
+  std::string compat_folder(arrays_dir + "/read_compatibility_test");
```

Comment:
NIT: We've seen bugs introduced this way... Let's avoid single line if/for/while... Always have the body between brackets.